onnxruntime/core/providers/nnapi/nnapi_builtin/selectors_actions/nnapi_qdq_selector_action_transformer.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/nnapi/nnapi_builtin/selectors_actions/nnapi_qdq_selector_action_transformer.h"
+#include "core/providers/nnapi/nnapi_builtin/selectors_actions/nnapi_qdq_selectors.h"
+
+namespace onnxruntime {
+
+using NTO = onnxruntime::ConstNodesToOptimize;
+
+inline void UnaryOpQDQRules(NNAPIQDQSelectorsAndActions& nnapi_qdq_selectors_and_actions) {
+  // 3 nodes. DQ, target, Q
+  // Replace with internal QLinear version of operator. Delete all original nodes.
+  const std::string action_name{"1DQ"};
+
+  std::unique_ptr<NNAPIQDQNodeSelector> selector(new NNAPIQDQ::UnarySelector());
```

Comment:
>selector [](http://example.com/codeflow?start=40&length=8)

general: consider using make_unique
`auto selector = std::make_unique<UnarySelector>();`
`unique_ptr<Derived>` is convertible to `unique_ptr<Base>`